src/controller/CHIPDeviceControllerFactory.h

```diff
@@ -138,6 +138,8 @@ class DeviceControllerFactory
     //
     void ReleaseSystemState() { mSystemState->Release(); }
 
+    const DeviceControllerSystemState * GetSystemState() { return mSystemState; }
```

Comment:
Needs some comments about usage:

_Retrieve a read-only pointer to the system state object that contains pointers to key stack singletons.  If the pointer is null, it indicates that the DeviceControllerFactory has yet to be initialized properly, or has already been shut-down.

This pointer ceases to be valid after a call to Shutdown has been made_